internal/dag/gatewayapi_processor.go

```diff
@@ -1976,6 +1981,74 @@ func (p *GatewayAPIProcessor) httpClusters(routeNamespace string, backendRefs []
 			continue
 		}
 
+		var upstreamValidation *PeerValidationContext
+		var backendRefGroup gatewayapi_v1alpha2.Group
+		if backendRef.Group != nil {
+			backendRefGroup = *backendRef.Group
+		}
+
+		var backendRefKind gatewayapi_v1alpha2.Kind
+		if backendRef.Kind != nil {
+			backendRefKind = *backendRef.Kind
+		}
+
+		policyTargetRef := gatewayapi_v1alpha2.PolicyTargetReferenceWithSectionName{
+			PolicyTargetReference: gatewayapi_v1alpha2.PolicyTargetReference{
+				Group:     backendRefGroup,
+				Kind:      backendRefKind,
+				Name:      backendRef.Name,
+				Namespace: backendRef.Namespace,
```

Comment:
this may not be specified if no cross-ns reference is present, so likely have to make this the ns of the HTTPRoute if empty

Reply:
Good catch. Done.